haystack/components/routers/transformers_text_router.py

```diff
@@ -0,0 +1,193 @@
+    from haystack.components.generators import HuggingFaceLocalGenerator
+
+    p = Pipeline()
+    p.add_component(
+        instance=TransformersTextRouter(model="papluca/xlm-roberta-base-language-detection"),
+        name="text_router"
+    )
+    p.add_component(
+        instance=PromptBuilder(template="Answer the question: {{query}}\nAnswer:"),
+        name="english_prompt_builder"
+    )
+    p.add_component(
+        instance=PromptBuilder(template="Beantworte die Frage: {{query}}\nAntwort:"),
+        name="german_prompt_builder"
+    )
+
+    p.add_component(
+        instance=HuggingFaceLocalGenerator(model="DiscoResearch/Llama3-DiscoLeo-Instruct-8B-v0.1"),
+        name="german_llm"
+    )
+    p.add_component(
+        instance=HuggingFaceLocalGenerator(model="google/gemma-1.1-7b-it"),
+        name="english_llm"
+    )
+
+    p.connect("text_router.en", "english_prompt_builder.query")
+    p.connect("text_router.de", "german_prompt_builder.query")
+    p.connect("english_prompt_builder.prompt", "english_llm.prompt")
+    p.connect("german_prompt_builder.prompt", "german_llm.prompt")
+
+    # English Example
+    print(p.run({"text_router": {"text": "What is the capital of Germany?"}}))
+
+    # German Example
+    print(p.run({"text_router": {"text": "Was ist die Hauptstadt von Deutschland?"}}))
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str,
+        device: Optional[ComponentDevice] = None,
+        token: Optional[Secret] = Secret.from_env_var("HF_API_TOKEN", strict=False),
+        huggingface_pipeline_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initializes the TransformersTextRouter.
+
+        :param model: The name or path of a Hugging Face model for text classification.
+        :param device: The device on which the model is loaded. If `None`, the default device is automatically
+            selected. If a device/device map is specified in `huggingface_pipeline_kwargs`, it overrides this parameter.
+        :param token: The API token used to download private models from Hugging Face.
+            If this parameter is set to `True`, the token generated when running
+            `transformers-cli login` (stored in ~/.huggingface) is used.
+        :param huggingface_pipeline_kwargs: Dictionary containing keyword arguments used to initialize the
+            Hugging Face pipeline for text classification.
+        """
+        torch_and_transformers_import.check()
+
+        self.token = token
+
+        huggingface_pipeline_kwargs = resolve_hf_pipeline_kwargs(
+            huggingface_pipeline_kwargs=huggingface_pipeline_kwargs or {},
+            model=model,
+            task="text-classification",
+            supported_tasks=["text-classification"],
+            device=device,
+            token=token,
+        )
+        self.huggingface_pipeline_kwargs = huggingface_pipeline_kwargs
+
+        config = AutoConfig.from_pretrained(
```

Comment:
After talking with @shadeMe I'll be moving this to `warm_up` to prevent requests happening in the init method.

Reply:
Actually we decided to resolve it by adding a `labels` param that is optional. If provided then it will skip the config fetching.